src/Elliptic/Systems/IrrotationalBns/FirstOrderSystem.hpp

```diff
@@ -0,0 +1,108 @@
+
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/DataBox/PrefixHelpers.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "Elliptic/BoundaryConditions/BoundaryCondition.hpp"
+#include "Elliptic/Protocols/FirstOrderSystem.hpp"
+#include "Elliptic/Systems/IrrotationalBns/Equations.hpp"
+#include "Elliptic/Systems/IrrotationalBns/Geometry.hpp"
+#include "Elliptic/Systems/IrrotationalBns/Tags.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp"
+#include "PointwiseFunctions/GeneralRelativity/TagsDeclarations.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace IrrotationalBns {
+
+/*!
+ * \brief The Irrotational Bns equations From Baumgarte and Shapiro Chapter 15
+ *  formulated as a set of coupled first-order PDEs.
+ *
+ * \details This system formulates the Irrotational Bns HSE equations for the
+ * velocity potential \f$\Phi\f$. For a background matter distribution (given
+ * by the specific enthalpy h) and a background metric \f$\gamma_{ij}\f$.
+ * The velocity potential is defined by \f$D_i \Phi = h u_i\f$ with \f$u_i\f$
+ * (the spatial part of) the four velocity
+ * and where \f$\Gamma^i_{jk}=\frac{1}{2}\gamma^{il}\left(\partial_j\gamma_{kl}
+ * +\partial_k\gamma_{jl}-\partial_l\gamma_{jk}\right)\f$ are the Christoffel
+ * symbols of the second kind of the background (spatial) metric
+ * \f$\gamma_{ij}\f$. The
+ * background metric \f$\gamma_{ij}\f$ and the Christoffel symbols derived from
+ * it are assumed to be independent of the variables \f$\Phi\f$ and \f$u_i\f$,
+ * i.e.
+ * constant throughout an iterative elliptic solve.  Additionally a background
+ * lapse (\f$\alpha\f$) and
+ * shift (\f$\beta\f$) must be provided.  Finally, a ``rotational killing
+ * vector" \f$k^i\f$ (with magnitude
+ * proportional to the angular velocity of the orbital motion) is provided.  The
+ * rotational shift is defined as \f$B^i = \beta^i + k^i\f$ which is
+ * heuristically the background
+ * motion of the spacetime.
+ *
+ * The system can be formulated in terms of these fluxes and sources (see
+ * `elliptic::protocols::FirstOrderSystem`):
+ *
+ *
+ * \f{align*}
+ * -\partial_i F^i + S = f
+ * \f}
+ *
+ * \f{align*}
+ * F^i &=  D_i \phi - \frac{B^j D_j \phi}{\alpha^2}B^i  \\
+ * S &= -F^iD_i \left( \ln \frac{\alpha \rho}{h}\right) -\Gamma^i_{ij}F^j \\
```

Comment:
Can you add spin terms already? This probably also means we should name the executable something other than "irrational".

Reply:
I'd like to have the irrotational case merged first, at that point it should be straightforward to extend to the spin case but also at that point things stop being "exact".  I'd like to go through the sources and understand what's going on and write up notes before I add that though. 

The minus sign is my bad I have confused myself thoroughly with signs.

Reply:
Ok that's fine. Probably still we should not name things "irrotational", don't you think? That seems confusing once we add the spin terms (and we don't want to rename everything later).